src/components/print/PrintDirective.js

```diff
@@ -623,13 +215,20 @@ goog.require('ga_urlutils_service');
       pdfLegendsToDownload = [];
       layersYears = [];
 
+      var dpi = getDpi($scope.layout.name, $scope.dpi);
+      var scaleDenom = parseFloat($scope.scale.value);
+      var printRectangeCoords = getPrintRectangleCoords();
+      var resolution = $scope.map.getView().getResolution();
+
+
       // Re order layer by z-index
       layers.sort(function(a, b) {
         return a.getZIndex() - b.getZIndex();
       });
 
       // Transform layers to literal
       layers.forEach(function(layer) {
```

Comment:
Any chance to have an `encodeLayers(layers)`  function ?  Could be in another PR.